apps/angular/projection/src/app/data-access/city.store.ts

```diff
@@ -1,23 +1,15 @@
-import { Injectable } from '@angular/core';
-import { BehaviorSubject } from 'rxjs';
+import { Injectable, effect } from '@angular/core';
+import { toSignal } from '@angular/core/rxjs-interop';
 import { City } from '../model/city.model';
+import { ShareStore } from './share.store';
 
 @Injectable({
   providedIn: 'root',
 })
-export class CityStore {
-  private cities = new BehaviorSubject<City[]>([]);
-  cities$ = this.cities.asObservable();
-
-  addAll(cities: City[]) {
-    this.cities.next(cities);
-  }
-
-  addOne(student: City) {
-    this.cities.next([...this.cities.value, student]);
-  }
-
-  deleteOne(id: number) {
-    this.cities.next(this.cities.value.filter((s) => s.id !== id));
+export class CityStore extends ShareStore<City> {
+  constructor() {
+    super();
+    const _cities = toSignal(this.http.fetchCities$, { initialValue: [] });
+    effect(() => this.addAll(_cities()), { allowSignalWrites: true });
```

Comment:
I think, it's not a good think to want to do everything with signals. 

Observable or Promise will not disappear, and having a subscribe on the http call is better than having this code. 

This code is hard to understand, less maintainable. 
if you write 

```ts
this.http.fetchCities().pipe(take(1)).subscribe(c => this.addAll(c))
```

This is easier to understand.

What do you think?

Reply:
I think the observable approach is often more familiar to any angular developers.
and the use of effect and toSignal is not that hard to understand I feel it's more understandable than using a lot of observable techniques. and we can avoid the subscription and let angular handle itself with signal approach. for maintainability, you can add/update/debug the code inside effect.

what do you think?

Reply:
for me, the code you wrote is mostly a hack of the signal API. Effect should be used as a last resort. Using `allowSignalWrites` is kind of a warning that this is not the way to go. 

I will always use observable or promise for http event. 

But if you feel like you prefer this approach, you can try it on bigger project and see how it goes. 😇

Reply:
yeah, you're right. allowSignalWrite is not recommended. Actually, I feel Signal is still under development by angular team. but I just try to get a better angular 17 signal approach instead of using the old rxjs technique. :)

Thank you for all suggestions.

Reply:
Yes effect is still experimental, but all the rest of the API is stable.